src/theory/arith/theory_arith_private.cpp

```diff
@@ -3938,15 +3938,36 @@ Node TheoryArithPrivate::branchIntegerVariable(ArithVar x) const {
   TNode var = d_partialModel.asNode(x);
   Integer floor_d = d.floor();
 
-  //Node eq = Rewriter::rewrite(NodeManager::currentNM()->mkNode(kind::EQUAL, var, mkRationalNode(floor_d+1)));
-  //Node diseq = eq.notNode();
+  if (options::cubeTest())
+  {
+    Trace("integers") << "cube test enabled"  << endl;
+    Integer ceil_d = d.ceiling();
+    Rational f = r - floor_d;
+    // Multiply by -1 to get abs value.
+    Rational c = (r - ceil_d) * (-1); 
+ 
+    Integer nearest;
+    (c > f) ? nearest = floor_d : nearest = ceil_d;
 
-  Node ub = Rewriter::rewrite(NodeManager::currentNM()->mkNode(kind::LEQ, var, mkRationalNode(floor_d)));
-  Node lb = ub.notNode();
+    Node ub = Rewriter::rewrite(NodeManager::currentNM()->mkNode(kind::LEQ, var, mkRationalNode(nearest - 1)));
+    Node lb = Rewriter::rewrite(NodeManager::currentNM()->mkNode(kind::GEQ, var, mkRationalNode(nearest + 1)));
+    Node lem = NodeManager::currentNM()->mkNode(kind::OR, ub, lb);
+    Node eq = Rewriter::rewrite(NodeManager::currentNM()->mkNode(kind::EQUAL, var, mkRationalNode(nearest)));
+    Node literal = d_containing.getValuation().ensureLiteral(eq);
+    d_containing.getOutputChannel().requirePhase(literal, true);
+    Node lem = NodeManager::currentNM()->mkNode(kind::OR, literal, lem);
+  }
+  else
+  {
+    //Node eq = Rewriter::rewrite(NodeManager::currentNM()->mkNode(kind::EQUAL, var, mkRationalNode(floor_d+1)));
+    //Node diseq = eq.notNode();
+    Node ub = Rewriter::rewrite(NodeManager::currentNM()->mkNode(kind::LEQ, var, mkRationalNode(floor_d)));
+    Node lb = ub.notNode();
```

Comment:
@amaleewilson i see this is not your code. Nevertheless, it seems wrong. 
It should have been `Node lb = mkNode(kind::GEQ, var, mkRationalNode(floor_d + 1))`

@ajreynol What do you think?

Reply:
I think these are equivalent.  x >= c+1 is equivalent to ~(x <= c ) right (for integers)?  I prefer this way since its clearer that this is indeed a proper split.

Reply:
> I think these are equivalent. x >= c+1 is equivalent to ~(x <= c ) right (for integers)?
I agree it is right for integers. However, the branch and bound lemmas are for LRA solver.
And actually we are not creating a split but removing a rational (non-integral) of the solution space.

For example, if LRA solver returned the solution x = 3.5 then the branch-and-bound lemma is going to tell the LRA solver that  x <=3 \/ x >= 4.

Reply:
The rewriter will take care of this.  Assuming x's type is integer, the rewriter will turn either constraint into the same thing (a normalized LEQ atom).

Reply:
i see. Then I agree with you.

Thanks for clarifying :)